panel/custom.py

```diff
@@ -182,17 +204,24 @@ def _init_params(self) -> dict[str, Any]:
                 params.pop(k)
             data_params[k] = v
         data_props = self._process_param_change(data_params)
+        precompiled = self._compiled_path is not None
+        if precompiled:
+            importmap = {}
+        else:
+            importmap = self._process_importmap()
         params.update({
             'class_name': camel_to_kebab(cls.__name__),
             'data': self._data_model(**{p: v for p, v in data_props.items() if p not in ignored}),
             'dev': config.autoreload or getattr(self, '_debug', False),
-            'esm': self._render_esm(),
-            'importmap': self._process_importmap(),
+            'esm': self._render_esm(not config.autoreload),
+            'importmap': importmap,
+            'precompiled': precompiled
         })
         return params
 
-    def _process_importmap(self):
-        return self._importmap
+    @classmethod
+    def _process_importmap(cls):
+        return cls._importmap
```

Comment:
This could now share instances between different objects is that ok?

Reply:
Yes, `_importmap` is defined as a class variable. Could make a copy here but it's a private API.